File: solutions/beecrowd/1030/1030.py
import sys


def josephus(n: int, k: int) -> int:
    result = 1
    for current_n in range(2, n + 1):
        result = (result + k - 1) % current_n + 1

    return result


_ = input()

i: int
line: str
for i, line in enumerate(sys.stdin):
    n: int
    k: int
    n, k = map(int, line.split())

    print(f'Case {i + 1}: {josephus(n, k)}')

File: solutions/beecrowd/1030/1030.clj
(ns main
  (:require [clojure.string :as str]))

(defn josephus [n k]
  (loop [current-n 2
         result 1]
    (if
     (> current-n n)
      result
      (recur (inc current-n)
             (-> result
                 (+ k)
                 dec
                 (mod current-n)
                 inc)))))

(defn main []
  (loop [nc (Integer/parseInt (read-line))
         index 1]
    (let [[n k] (->> (str/split (read-line) #" ")
                     (map #(Integer/parseInt %)))]
      (printf "Case %d: %d%n" index (josephus n k))
      (when (< index nc)
        (recur nc (inc index))))))

(main)
